Design/spectrum_number_and_workspace_index_abstraction.md

```diff
@@ -0,0 +1,269 @@
+# Indexing Abstraction Layer
```

Comment:
Nice diagram. This is the only place where "spectrum index" appears. Maybe that needs to be explained somewhere?
It seems it is not a Mantid concept, but as that wording is used in NeXus files it is probably good to have its meaning clearly defined here. 
I'm not totally sure if "spectrum index" is equivalent to "workspace index". Does it always increase by one? Is that stated/guaranteed anywhere?

I think that increasing by one has been the common practice at ISIS, but recently we have seen files where they break that assumption for some prototype detectors.